### PR TITLE
If glog source info exists, parse as expected by the output plugin.

### PR DESCRIPTION
**Pull request by jkschulz**

### CODE REVIEW

File: agents.yaml [hunk 1]
```diff
@@ -610,6 +610,22 @@ data:
...
+    #     {"source":"handlers.go:131"},
+    # then the following entry will be added to the record:
+    #     {"logging.googleapis.com/sourceLocation":
+    #          {"file":"handlers.go", "line":"131"}
+    #     }
+    <filter **>
+      @type record_transformer
+      enable_ruby true
+      <record>
+        "logging.googleapis.com/sourceLocation" ${if record.has_key?('source'); source_parts = record['source'].split(':', 2); {'file' => source_parts[0], 'line' => source_parts[1]}; else; nil; end}
```
Comment (bmoyles0117) [inline agents.yaml]: Let's add an additional check at the end when setting the hashmap `{'file' => source_parts[0], 'line' => source_parts[1]} if source_parts.length == 2`

Comment (jkschulz) [reply agents.yaml]: Done

File: agents.yaml [hunk 2]
```diff
@@ -610,6 +610,22 @@ data:
...
+    #     {"source":"handlers.go:131"},
+    # then the following entry will be added to the record:
+    #     {"logging.googleapis.com/sourceLocation":
+    #          {"file":"handlers.go", "line":"131"}
+    #     }
+    <filter **>
+      @type record_transformer
+      enable_ruby true
+      <record>
+        "logging.googleapis.com/sourceLocation" ${if record.has_key?('source'); source_parts = record['source'].split(':', 2); {'file' => source_parts[0], 'line' => source_parts[1]} if source_parts.length == 2; else; nil; end}
```
Comment (qingling128) [inline agents.yaml]: If `resource` is nil, `record['source'].split` might error out? (Not sure how likely that is...)

```
$ irb
2.4.3 :001 > nil.split(':', 2)
NoMethodError: undefined method `split' for nil:NilClass
        from (irb):1
        from /usr/local/google/home/lingshi/.rvm/rubies/ruby-2.4.3/bin/irb:11:in `<main>'
```

Comment (jkschulz) [reply agents.yaml]: From https://github.com/fluent/fluentd/issues/176, it sounds like this _shouldn't_ happen, but it's possible so I'll add the check.

**Pull request merged**